src/pages/[locale]/products.tsx

```diff
@@ -0,0 +1,182 @@
+            This page uses server side rendering (SSR)
+
+            Each page refresh (either SSR or CSR) queries the GraphQL API and displays products below:
+          `}
+        </Text>
+
+        <hr />
+
+        <h2>Published products</h2>
+
+        <Products
+          products={productsPublished}
+        />
+
+        <hr />
+
+        <h2>Draft products</h2>
+
+        <Text>
+          {`
+            Those products are being created/updated by the NRN community, anybody can manipulate those through <a href="https://nrn-admin.now.sh/#/Product/create" target="_blank">the Admin site</a>.
+
+            Don't hesitate to give it a try, you'll see the list of products below will update because content is fetched for every page request.
+          `}
+        </Text>
+
+        <Products
+          products={productsDraft}
+        />
+      </Container>
+    </Layout>
+  );
+};
+
+ProductsPage.getInitialProps = async (context: NextPageContext & { apolloClient: ApolloClient<NormalizedCacheObject> }): Promise<Props> => {
+  const {
+    apolloClient,
+    AppTree,
+    asPath,
+    err,
+    query,
+    pathname,
+    req,
+    res,
+  } = context;
+  const customerRef: string = process.env.CUSTOMER_REF;
+  const readonlyCookies: Cookies = NextCookies(context); // Parses Next.js cookies in a universal way (server + client)
+  const cookiesManager: UniversalCookiesManager = new UniversalCookiesManager(req, res);
+  const userSession: UserSemiPersistentSession = cookiesManager.getUserData();
+  const { headers }: IncomingMessage = req;
+  const publicHeaders = {
+    'accept-language': get(headers, 'accept-language'),
+    'user-agent': get(headers, 'user-agent'),
+    'host': get(headers, 'host'),
+  };
+  const hasLocaleFromUrl = !!query?.locale;
+  const locale: string = hasLocaleFromUrl ? query?.locale as string : DEFAULT_LOCALE; // If the locale isn't found (e.g: 404 page)
+  const lang: string = locale.split('-')?.[0];
+  const bestCountryCodes: string[] = [lang, resolveFallbackLanguage(lang)];
+  const gcmsLocales: string = prepareGraphCMSLocaleHeader(bestCountryCodes);
+  const defaultLocales: I18nextResources = await fetchTranslations(lang); // Pre-fetches translations from Locize API
+  const variables = {
+    customerRef,
+  };
+  const queryOptions = {
+    displayName: 'PRODUCTS_PAGE_QUERY',
+    query: PRODUCTS_PAGE_QUERY,
+    variables,
+    context: {
+      headers: {
+        'gcms-locale': gcmsLocales,
+      },
+    },
+  };
+
+  const {
+    data,
+    errors,
+    loading,
+    networkStatus,
+    stale,
+  }: ApolloQueryResult<{
+    customer: Customer;
+    products: Product[];
+  }> = await apolloClient.query(queryOptions);
+
+  if (errors) {
+    console.error(errors);
+    throw new Error('Errors were detected in GraphQL query.');
+  }
+
+  const {
+    customer,
+    products,
+  } = data || {}; // XXX Use empty object as fallback, to avoid app crash when destructuring, if no data is returned
+
+  return {
```

Comment:
Not sure why you're exporting all those props, are they really required in the page component? eg: `headers`, `hasLocaleFromUrl`, `lang`.

Asking because it's a little bit confused to understand where they're going to be used/required, maybe adding some docs/comments.

Reply:
Ok, I've noticed that a few of those props are being for Amplitude and GraphCMS.

Reply:
Yeah, they're needed. Headers is unused but could be handy for various reasons. hasLocaleFromUrl can help dev to know how the language was resolved (handy when using multiple resolvers, like cookies and such), lang is required for the app to work, bestCountryCodes is for graphcms GQL queries that may happen in the components (could be handy to have access to it, even though it's not currently useful once the queries are performed)

Reply:
Yeah, make sense now, maybe add that information on `getCommonServerSideProps` would be super helpful.

Reply:
Just did, let me know if it should be improved. Closing